hyperlink/test/test_scheme_registration.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+
+from .common import HyperlinkTestCase
+from .._url import register_scheme, URL
+
+
+class TestSchemeRegistration(HyperlinkTestCase):
+
+    def test_register_scheme_basic(self):
+        register_scheme('deltron', uses_netloc=True, default_port=3030)
```

Comment:
This modifies [global state inside `hyperlink`](https://github.com/python-hyper/hyperlink/blob/master/hyperlink/_url.py#L328-L334), but it doesn't look like there's anything that will unwind this change after this test's completion.  That's a recipe for trouble down the way.

It looks like a `setUp` method could copy `hyperlink._url.SCHEME_PORT_MAP` and `hyperlink._url.NO_NETLOC_SCHEMES`, which a `tearDown` method could restore.  Please add something like this to minimize the chances of unexpected dependencies between tests.